internal-enrichment/shodan/src/shodanImport.py

```diff
@@ -332,27 +338,50 @@ def _generate_stix_vulnerability(self, data):
 
     def _upsert_stix_observable(self, description, labels, external_reference):
         # Upsert Observable
-        stix_observable = stix2.IPv4Address(
-            id=self.stix_entity["id"],
-            type="ipv4-addr",
-            value=self.stix_entity["value"],
-            custom_properties={
-                "x_opencti_external_references": [external_reference],
-                "x_opencti_description": description,
-                "x_opencti_score": self.score,
-                "x_opencti_labels": labels,
-                "x_opencti_created_by_ref": self.shodan_identity["standard_id"],
-            },
-        )
-        self.stix_objects.append(stix_observable)
-        """
-        {
-                    "source_name": "Shodan",
-                    "url": f"www.shodan.io/host/{data['ip_str']}",
-                    "description": "ceci est une description",
-                    # "external_id": str(data['ip_str']),
-                }
-        """
+        if not self.create_note:
+            stix_observable = stix2.IPv4Address(
+                id=self.stix_entity["id"],
+                type="ipv4-addr",
+                value=self.stix_entity["value"],
+                custom_properties={
+                    "x_opencti_external_references": [external_reference],
+                    "x_opencti_description": description,
+                    "x_opencti_score": self.score,
+                    "x_opencti_labels": labels,
+                    "x_opencti_created_by_ref": self.shodan_identity["standard_id"],
+                },
+            )
+            self.stix_objects.append(stix_observable)
+            """
+            {
+                        "source_name": "Shodan",
+                        "url": f"www.shodan.io/host/{data['ip_str']}",
+                        "description": "ceci est une description",
+                        # "external_id": str(data['ip_str']),
+                    }
+            """
+        else:
+            # Create note instead of overwriting description
+            stix_observable = stix2.IPv4Address(
+                id=self.stix_entity["id"],
+                type="ipv4-addr",
+                value=self.stix_entity["value"],
+                custom_properties={
+                    "x_opencti_external_references": [external_reference],
+                    "x_opencti_score": self.score,
+                    "x_opencti_labels": labels,
+                    "x_opencti_created_by_ref": self.shodan_identity["standard_id"],
+                },
+            )
+            self.stix_objects.append(stix_observable)
+            stix_note = stix2.Note(
```

Comment:
And put the condition for the note directly, small attention can you add the generate_id for the note ?
```Python
# import 
from pycti import Note

# Example
if not self.create_note:
  now = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
  stix_note = stix2.Note(
              type="note",
              id=Note.generate_id(now, description),
              abstract="Shodan Results",
              content=description,
              created_by_ref=self.shodan_identity["standard_id"],
              object_refs=self.stix_entity["id"],
          )
          self.stix_objects.append(stix_note)
```